src/lightning/pytorch/strategies/xla.py

```diff
@@ -134,6 +128,12 @@ def distributed_sampler_kwargs(self) -> Dict[str, int]:
     def is_distributed(self) -> bool:
         # HOST_WORLD_SIZE is not set outside the xmp.spawn process
         import torch_xla.core.xla_env_vars as xenv
+        from torch_xla.experimental import pjrt
+
+        if pjrt.using_pjrt():
+            from multiprocessing import current_process
+
+            return current_process().name != "MainProcess" and self.world_size != 1
```

Comment:
PJRT don't set `HOST_WORLD_SIZE` env var, so here we check the process name to check if it is in a spawned process.

Reply:
We could replace the process name check with `self._launched`, no?

Reply:
`self._launched` is set in the middle of `trainer._run()`, so if `self.is_distributed` is not used before that point, I guess that would be okay. For fabric, I guess it is similar?

Reply:
Opened https://github.com/Lightning-AI/lightning/pull/17381